### PR TITLE
Add `nix run` and `nix build`

### PR DESCRIPTION
**Pull request by srid**
This is largely the work of @MatthewCroughan to enable `nix run` to be used to quickly run the website locally with hot reload, and `nix build` to generate the static website.

This PR should be squash merged, since there are a lot commits.

### CODE REVIEW

**Comment by srid** [comment]
new README view: https://github.com/Plutonomicon/plutonomicon/tree/srid/mv

**Review by MatthewCroughan — CHANGES_REQUESTED** [review]
@srid 

Can you not add an `apps` attribute to the flake so that you can simply run `nix run .#live` from this flake instead of having to add another file/folder `(bin/run)`to this repo? Would you accept me making this change as a commit on top of what you've done?

**Comment by srid** [comment]
> Would you accept me making this change as a commit on top of what you've done?

Yes, thank you, @MatthewCroughan

**Comment by MatthewCroughan** [comment]
I've rebased this branch on `main` in order to allow myself to make the change, since the `flake.nix` has been updated in the meantime.

**Comment by srid** [comment]
@MatthewCroughan Okay, but please no rebase in future for multi-author branches; you can just `git merge origin/main`. We can always squash merge the PR for cleaner history `main`.

File: flake.nix [hunk 1]
```diff
@@ -17,6 +17,17 @@
       hci-effects = hercules-ci-effects.lib.withPkgs pkgs;
     in
     {
+      apps."${system}" = {
+        live = rec {
+          type = "app";
+          script = pkgs.writers.writeBash "emanotePlutonomiconLiveReload.sh" ''
+            set -xe
+            export PORT=7071
```
Comment (srid) [inline flake.nix]: Does `nix run` pass the user environment? If so, we would want to use it, and fallback to 7071 only if no $PORT is set. It allows user to change the port number if they desire.

Comment (srid) [reply flake.nix]: Not critical for this PR though.

**Comment by srid** [comment]
Might as well make `live` the defaultApp, so that `nix run .` works too?

File: flake.nix (hunk 1 above)
Comment (MatthewCroughan) [reply flake.nix]: `PORT` seems to be a bit general. Can you not make Emanote take `EMANOTE_PORT` instead? I can imagine `PORT` being set by a lot of things, for whatever reason.

Comment (srid) [reply flake.nix]: > I can imagine `PORT` being set by a lot of things, for whatever reason.

Can you elaborate?

Comment (srid) [reply flake.nix]: Or better, I should make it take it as argument, along with `$HOST`: `emanote run -b 127.0.0.1:7071`

Comment (MatthewCroughan) [reply flake.nix]: I've implemented this in e7272cb18da8dc2be50870bc1e3f5dae0b36481f

**Comment by MatthewCroughan** [comment]
> Might as well make `live` the defaultApp, so that `nix run .` works too?

Done in `e20aada56b40b2bf56e6aa32d620e088e3864a15`

File: flake.nix (hunk 1 above)
Comment (srid) [reply flake.nix]: > Or better, I should make it take it as argument, along with `$HOST`: `emanote run -b 127.0.0.1:7071`

Done in https://github.com/srid/emanote/commit/cb8c989bd39563ad14cb569f1c9e272efd5c5c31

File: .vscode/extensions.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+	// See https://go.microsoft.com/fwlink/?LinkId=827846
+	// for the documentation about the extensions.json format
+	"recommendations": [
+		"yzhang.markdown-all-in-one",
+		"svsool.markdown-memo",
+		"bierner.markdown-checkbox",
```
Comment (L-as) [inline .vscode/extensions.json]: What is this?

Comment (srid) [reply .vscode/extensions.json]: It's https://marketplace.visualstudio.com/items?itemName=bierner.markdown-checkbox

Removed now, along with a few others; they came from https://github.com/srid/emanote-template/tree/master/.vscode

**Pull request merged**